### PR TITLE
Graph theory - minor improvements of Huneke's proof (7)

### PR DESCRIPTION
**Pull request by avekens**
* ~cofmpt moved from TA's mathbox to main set.mm
* ~reximdvva added. The proofs of the following 8 theorems could be shortened: ~lcmgcdlem, ~lsmelval2, ~ftc1anclem6, ~eulerpartlemgvv, ~cvmlift2lem10, ~cpmadugsum, ~axpasch, ~frgrwopreglem5
* ~frgrwopregasn, ~frgrwopregbsn, ~frgrwopreglem4a, ~frgrwopreglem5a, ~frgrwopreglem5lem added
* proofs of ~frgrwopreg1, ~frgrwopreg2, ~frgrwopreglem4, ~frgrwopreglem5 shortened
* Some changes from review of PR [#2474](https://github.com/metamath/set.mm/issues/2474)

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -424459,6 +424480,51 @@ then they have the same degree (i.e., the same number of adjacent
       BVSURUSUTVAVBUNSWJWAVFWJVQWAVQWIULTWSRVCVDWKDVGJZWGPZWOWMOWJXFWAVQXEWIWGD
       VOZWTVHTVSDEFVIVJWKXEXAPZWQWNOWJXHWAVQXEWIXAXGXBVHTVRDEFVIVJVKVSCWLGVLVRC
       WLGVLVMVPVN $.
+
+    $d D x y $.  $d X x y $.  $d Y x y $.
+    frgrwopreglem4a.e $e |- E = ( Edg ` G ) $.
+    $( In a friendship graph each vertex is connected with a vertex with a
+       different degree.  Alternate version of ~ frgrwopreglem4 without a fixed
```
Comment (benjub) [inline set.mm:424487]: The statement looks more powerful: "...each vertex is connected to **any** vertex of different degree" or maybe better "any two vertices with different degrees are connected".

File: set.mm [hunk 2]
```diff
@@ -61941,10 +61953,9 @@ the restriction (of the relation) to the singleton containing this
     ( wrel cres cdm cin resdm resindi incom inres inidm reseq1i 3eqtrri 3eqtr4g
     ineq2d ) ACZABDZAAEZDZFQAFZABRFDQPSAQAGOABRHTAQFAAFZBDQQAIAABJUAABAKLMN $.
 
-  $( Restricting a function to a domain without one element of the domain of
-     the function, and adding a pair of this element and the function value of
-     the element results in the function itself.  (Contributed by AV,
-     2-Dec-2018.) $)
+  $( Restricting a relation to its domain without one element of the domain is
```
Comment (benjub) [inline set.mm:61956]: remove "of the domain"

File: set.mm [hunk 3]
```diff
@@ -424459,6 +424480,51 @@ then they have the same degree (i.e., the same number of adjacent
...
+      NZXFANZPZQZJDXGUJZUKZULKDULZWOWRXBMZXEQKJACDGHUMXQXPWOXEXQXPFCESTZUIZXDQZ
+      WOXEQZXQXTXFXRUIZWSXKPZQXMKJEFDXODEUJZUKZXGEPZXIYBXLYCYFXHXRPXIYBUNXGECSU
+      SXHXRXFUOUPYFXJWSXKXGEARUQURXFFPZYBXSYCXDXFFXRUTYGXKWTWSXFFARVAURWPWQXBVB
+      YFXOYEPXQYFXNYDDXGEVCVDVEXQWQFEOZMFYELWRWQXBYHWPWQVFXBYHEFVGVHVIFDEVJVKVL
+      XTXQYAXSXDXQYAQZXSVMFXRLZYIFXRVNYJYAXQWOYJXEWOYJMXCXAWOYJXCWOYJECFSTLZXCC
+      EFUGVOWOYKXCWOCVTLYKXCUNCVPBCFEIVQUPVRVSUEWAWBWAWCXDXEXQWOXCWSWTWDWEWFWGW
+      HWIWJWKWLWMWN $.
+
+    $( If a friendship graph has two vertices with the same degree and two
+       vertices with different degrees, then there is a 4-cycle in the graph.
```
Comment (benjub) [inline set.mm:424509]: "... with the same degree and two other vertices with degrees different from that common degree, then these four vertices form a 4-cycle."

File: set.mm [hunk 4]
```diff
@@ -424505,130 +424571,196 @@ then they have the same degree (i.e., the same number of adjacent
...
+      $d A b $.  $d B x $.  $d D y $.  $d G a b x y $.  $d V y $.
+      frgrwopreg.e $e |- E = ( Edg ` G ) $.
+      $( Lemma 4 for ~ frgrwopreg .  In a friendship graph each vertex with
+         degree ` K ` is connected with a vertex with degree other than ` K ` .
+         This corresponds to statement 4 in [Huneke] p. 2:  "By the first
+         claim, every vertex in A is adjacent to every vertex in B.".
+         (Contributed by Alexander van der Vekens, 30-Dec-2017.)  (Revised by
+         AV, 10-May-2021.)  (Proof shortened by AV, 4-Feb-2022.) $)
+      frgrwopreglem4 $p |- ( G e. FriendGraph
+                             -> A. a e. A A. b e. B { a , b } e. E ) $=
```
Comment (benjub) [inline set.mm:424584]: I can't see the $e hypotheses. Probably A (resp. B) is defined as the set of vertices with degree K (resp. different from K) ? Then, the statement is stronger than what the comment indicates: each vertex with degree K is connected to **any** vertex with degree different from K"

**Review by benjub — APPROVED** [review]
with a few comments

File: set.mm (hunk 2 above)
Comment (benjub) [reply set.mm:61956]: ok

File: set.mm (hunk 3 above)
Comment (avekens) [reply set.mm:424509]: I will change that comment slightly...

**Review by jkingdon — APPROVED** [review]

**Pull request merged**